### ISSUE TITLE
LaunchPad on atomic site preview - "Coming Soon"

### ISSUE THREAD

**Issue by Addison-Stavlo** [Automattic/wp-calypso#73377]
There seems to be some regression in our coming soon bypass for atomic in the launchpad preview (previous art in https://github.com/Automattic/wp-calypso/pull/71636) . Visit the launchpad on an atomic site (business plan -> activate hello dolly plugin), after transfer visit my home and get redirected to the launchpad. Notice 'coming soon' instead of the site preview.

UPDATE: I spent several hours testing and looking at this today. To refine the above findings: 
   - This issue appears to be intermittent. I created about 25-30 new test sites today, and transferred them to Atomic. I would say I saw this issue about 1 out of 4 times. The other times the web preview loaded correctly, with 
      - About 3-4 times, I experienced a situation where the launchpad_screen option was false on atomic, and the site would not redirect to launchpad at all. This is a separate issue, and also appears to be intermittent. 
   - This issues resolves by refreshing the launchpad screen. So when you see it, if you just refresh the page, it goes away. 
   - I assume this issue is related to our use of share codes to bypass the coming soon screen on atomic. See this PR. However, I found that even when I saw the coming soon page, a share code did appear to be generated and appended to the url of the web preview. The share code value would initially be undefined, but would ultimately be populated. Despite that, the coming soon screen loaded. 
   - If you start to troubleshoot the share code issue, search for `is_accessed_by_valid_share_link` as a starting point. You should find it the jetpack-mu-plugin under features/coming-soon. 

![image](https://user-images.githubusercontent.com/28742426/219057978-62a0985d-59f5-4d9d-9253-6bc215875660.png)

### Steps to reproduce

1. Start at https://wordpress.com/start/domains and create a new site.
2. When you get plans, choose a Business plan. On the Goals screen, chose 'Promote myself or my business'.
3. Continue through setup until you get to first Launchpad screen.
4. Click Skip to Dashboard, and activate any plugin. This will initiate the Atomic transfer. 
5. Once plugin is installed and atomic transfer is done, navigate to My Home. It should redirect you to launchpad. 
    - If you are not redirected to home, just enter the launchpad screen url manually. (https://wordpress.com/setup/build/launchpad?siteSlug=YOURSITESLUG). But note that every time I had this issue and went directly to launchpad, I could not duplicate the issue with the Coming Soon screen showing. 

### What I expected

When you get to launchpad, the web preview should show the actual site with your theme.

### What happened instead

What you will see instead - in some cases - is the coming soon screen.

**Comment by ash1eygrace** [comment]
Just a note I ran into an issue on a ping p1724499194310919-slack-C03TY6J1A.
User ticket: 8646084-zd-a8c

When accessing WordPress.com the users site was redirecting to:

https://wordpress.com/setup/false/goals?siteSlug=wordsatnight.com (just a white screen) 

To fix it, I manually went to: 
https://wordpress.com/setup/build/launchpad?siteSlug=YOURSITESLUG

That seemed to break the site out of the redirect to https://wordpress.com/setup/false/goals?siteSlug=wordsatnight.com

It also, seems related to: https://github.com/Automattic/wp-calypso/issues/88384 

The users's case above is fixed, but just noting this incase it helps.

**Comment by richtabor** [comment]
This launchpad is being deprecated for an in-WordPress launchpad in https://github.com/Automattic/wp-calypso/issues/98245. I'll go ahead and close this.
